cirq/contrib/qcircuit/qcircuit_test.py

```diff
@@ -12,25 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import itertools
+
 import cirq
-from cirq.contrib import circuit_to_latex_using_qcircuit
-from cirq.contrib.qcircuit.qcircuit_diagram import _QCircuitQubit
-from cirq.contrib.qcircuit.qcircuit_diagrammable import (
-    _TextToQCircuitDiagrammable
-)
-
-
-def test_qcircuit_qubit():
-    x = cirq.NamedQubit('x')
-    qx = _QCircuitQubit(x)
-    qx2 = _QCircuitQubit(x)
-    qy = _QCircuitQubit(cirq.NamedQubit('y'))
-    assert repr(qx) == '_QCircuitQubit({!r})'.format(x)
-    assert x != qx
-    assert qx != 0
-    assert qx == qx2
-    assert qx != qy
-    assert sorted([x, qy, qx]) == [x, qx, qy]
+import cirq.contrib.qcircuit as ccq
+
+
+def assert_qcircuit_diagrams_equal(actual: str, expected: str):
+    zipped_lines = itertools.zip_longest(
+            actual.split(r'\\'), expected.split(r'\\'), fillvalue='')
+    for line_number, (actual_line, expected_line) in enumerate(zipped_lines):
```

Comment:
`cirq.testing.highlight_text_differences`

Reply:
In this case, it's not necessary that the strings be exactly the same, because the "diagrams" are really the code to produce the diagrams, so most whitespace is irrelevant. This actually strips the entries of the matrix. It's not perfect, because it doesn't account for escaped ampersands, but it should work even when they're present. It also doesn't normalize whitespace within the entries. A few possibilities:

* Just test exact equality of the strings and use `highlight_text_differences`. This would be the prettiest by overly exacting.
* Test for equality in the current way, but use `highlight_text_differences` in reporting. Easiest way but a little confusing.
* Extend `highlight_text_differences` to split each line and strip surrounding whitespace of entries.

Reply:
Whitespace is sometimes significant in latex and sometimes not. E.g. "\times a" is not the same thing as "\timesa". I also think of these tests as "I want it to be exactly this; the alignment is important" tests.

Reply:
Done.